src/DataManagement/MonthlyReport/MonthlyReportCalculator.cs

```diff
@@ -0,0 +1,136 @@
+            var incomeTotal = Income.GetSum();
+            var thisMonthExpenses = Expenses.GetFrom(StartTime).GetTo(EndTime).GetSum();
+            var thisMonthIncome = Income.GetFrom(StartTime).GetTo(EndTime).GetSum();
+            AllSavings = incomeTotal - expensesTotal;
+            Balance = thisMonthIncome - thisMonthExpenses;
+
+            return new MonthlyReportResult(thisMonthExpenses, thisMonthIncome, Balance, StartTime, EndTime);
+
+        }
+
+        private void CalculateBiggestExpensesCategory()
+        {
+            var enumCount = Enum.GetValues(typeof(Importance)).Length;
+            var biggestCategory = (Importance) enumCount;
+            var sum = 0m;
+            foreach (var importance in Enum.GetValues(typeof(Importance)).Cast<Importance>())
+            {
+                var temp = 0M;
+                var expenses = Expenses.GetBy(importance);
+                foreach (var entry in expenses) //not linq since when empty category it crashes
+                {
+                    temp += entry.Amount;
+                }
+                if (temp < sum) continue;
+                sum = temp;
+                biggestCategory = importance;
+            }
+
+            Result.BiggestCategory = biggestCategory;
+            Result.BiggestCategorySum = sum;
+            var necessarySum = Expenses.GetBy(Importance.Necessary).Sum(entry => entry.Amount);
+            Result.NecessarySum = necessarySum;
+            if (necessarySum == 0)
+            {
+                Result.HowMuchBigger = -1;
+                return;
+            }
+            Result.HowMuchBigger = ((sum - necessarySum) / necessarySum * 100);
+        }
+
+        private List<IGoal> CalculateSavedUpGoals()
+        {
+            return AllSavings >= 0 ? Goals.Where(x => x.Amount <= AllSavings).ToList() : new GoalList();
+        }
+
+        private void CalculateGoalReport()
+        {
+            if (Goals.Count < 2 || Balance <= 0)
+            {
+                Result.HasGoals = false;
+                return;
+            }
+
+            Result.HasGoals = true;
+            var min = decimal.MaxValue;
+            var minGoal = new Goal();
+            var maxGoal = new Goal();
+            var max = 0M;
+            foreach (var item in Goals)
+            {
+                if (item.Amount >= max)
+                {
+                    max = item.Amount;
+                    maxGoal = (Goal)item;
+                }
+
+                if (item.Amount > min || item.Amount <= AllSavings ) continue;
+                min = item.Amount;
+                minGoal = (Goal) item;
+            }
+
+            Result.CheapestGoal = minGoal;
+            Result.MostExpensiveGoal = maxGoal;
+
+            Result.MonthsForCheapestGoal = (int) decimal.Ceiling((minGoal.Amount - AllSavings) / Balance);
+            Result.MonthsForMostExpensiveGoal = (int)decimal.Ceiling((maxGoal.Amount - AllSavings) / Balance);
```

Comment:
The formatting is a bit inconsistent here, quick auto format would help

Reply:
Could you elaborate?

Reply:
Let's say the lines:

            Result.MonthsForCheapestGoal = (int) decimal.Ceiling((minGoal.Amount - AllSavings) / Balance);
            Result.MonthsForMostExpensiveGoal = (int)decimal.Ceiling((maxGoal.Amount - AllSavings) / Balance);

The cast in the first line has a space afterwards, and in the second line doesn't have it. You can auto format in visual studio with the shortcut Ctrl+K, Ctrl+D, this formats the whole document :D